### PR TITLE
completed task - 3

### PR DESCRIPTION
**Pull request by Abdullazadeh-Nasir**

### CODE REVIEW

File: src/Task3/Family.java [hunk 1]
```diff
@@ -0,0 +1,167 @@
+package Task3;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+public class Family extends Human {
+
+    static {
```
Comment (AgilAghamirzayev) [inline src/Task3/Family.java:8]: do we need it?

File: src/Task3/Family.java [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    public void setFather(Human father) {
+        this.father = father;
+        countFamily++;
+    }
+
+    public Human[] getChild() {
+        return children;
+    }
+
+    public void setChild(Human[] child) {
```
Comment (AgilAghamirzayev) [inline src/Task3/Family.java]:
```
    public void setChild(Human child) {

 }
```

write like that

File: src/Task3/Family.java [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+                c=child;
+                System.out.println("The child was added to the family");
+                break;
+            }
+            else {
+                isEmpty=true;
+            }
+        }
+
+        Human [] c=new  Human[children.length+1];
```
Comment (AgilAghamirzayev) [inline src/Task3/Family.java]: Use Arrays.clone() method

File: src/Task3/Family.java [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        boolean isEmpty=false;
+        countFamily++;
+        for(Human c: children){
+            if(c==null){
+                c=child;
+                System.out.println("The child was added to the family");
+                break;
+            }
+            else {
+                isEmpty=true;
```
Comment (AgilAghamirzayev) [inline src/Task3/Family.java]: Use array indexing

File: src/Task3/Family.java [hunk 5]
```diff
@@ -0,0 +1,167 @@
...
+                c[children.length]=child;
+                System.out.println("The child was added to the family");
+                children=c;
+            }
+        }
+    }
+
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (AgilAghamirzayev) [inline src/Task3/Family.java:137]: Good

File: src/Task3/Human.java [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+
+    public String getName() {
+        return name;
+    }
+
+    public  void setName(String name) {
+        String  regex="[a-zA-Z]";
+        StringBuilder result=new StringBuilder();
+
```
Comment (AgilAghamirzayev) [inline src/Task3/Human.java:29]: bool isValid = name.match(regex);

**Review by AgilAghamirzayev — COMMENTED** [review]
Good

**Comment by Abdullazadeh-Nasir** [comment]
Teacher problems were eliminated :) Can I merge?

**Pull request merged**